### PR TITLE
tasks.py: add update-terraform, terraform/flake.lock: Update

### PR DESCRIPTION
**Pull request by zowoq**
```
Terraform updates:

terraform: 1.3.9 → 1.4.0
terraform-provider-cloudflare: 3.33.1 → 4.1.0
terraform-provider-external: 2.2.3 → 2.3.1
```
https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/guides/version-4-upgrade

I'd appreciate someone double checking that the upgrade is okay.

### CODE REVIEW

**Comment by zowoq** [comment]
bors try

File: tasks.py [hunk 1]
```diff
@@ -198,6 +198,25 @@ def scan_age_keys(c, host):
...
+    c.run(
+        """
+system="$(nix eval --impure --raw --expr 'builtins.currentSystem')"
+old="$(nix build --no-link --print-out-paths ".#devShells.${system}.default")"
+nix flake update --commit-lock-file
+new="$(nix build --no-link --print-out-paths ".#devShells.${system}.default")"
+commit="$(git log --pretty=format:%B -1)"
+diff="$(nix store diff-closures "${old}" "${new}" | awk -F ',' '/terraform/ && /→/ {print $1}')"
+git commit --amend -m "${commit}" -m "Terraform updates:" -m "${diff}"
+"""
```
Comment (Mic92) [inline tasks.py:216]:
```suggestion
    with c.cd("terraform"):
        c.run("""
system="$(nix eval --impure --raw --expr 'builtins.currentSystem')"
old="$(nix build --no-link --print-out-paths ".#devShells.${system}.default")"
nix flake update --commit-lock-file
new="$(nix build --no-link --print-out-paths ".#devShells.${system}.default")"
commit="$(git log --pretty=format:%B -1)"
diff="$(nix store diff-closures "${old}" "${new}" | awk -F ',' '/terraform/ && /→/ {print $1}')"
git commit --amend -m "${commit}" -m "Terraform updates:" -m "${diff}"
        """)
```

Comment (Mic92) [reply tasks.py:216]: https://docs.pyinvoke.org/en/stable/api/context.html#invoke.context.Context.cd

Comment (zowoq) [reply tasks.py]: Done.

**Comment by Mic92** [comment]
bors merge

**Pull request merged**